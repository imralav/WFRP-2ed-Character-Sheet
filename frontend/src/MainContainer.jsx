import React, {Fragment, Component} from 'react';
import {Route} from 'react-router-dom';

import OpenCharacter from './open-character';
import TranslationsProvider from './translations/TranslationsProvider';
import CharacterSheet from './character-sheet';
import ModeSelector from './mode-selector';
import Header from './header';

import 'bootstrap/dist/css/bootstrap.css'
import './index.css';

import pl from 'react-intl/locale-data/pl';
import en from 'react-intl/locale-data/en';
import {addLocaleData, IntlProvider} from 'react-intl';

addLocaleData(pl);
addLocaleData(en);


export default class MainContainer extends Component {
    constructor(props) {
        super(props);
        this.state = {
            locale: 'pl',
            messages: TranslationsProvider.get()
        }
    }

    changeLocale(locale) {
        this.setState({...this.state, locale});
    }

    render() {
        return (
            <IntlProvider locale={this.state.locale} messages={this.state.messages[this.state.locale]}>
                <Fragment>
                    <div className="container-fluid">
                        <Header changeLocale={locale => this.changeLocale(locale)}/>
                        <Route exact path='/' component={ModeSelector}/>
                        <Route exact path='/character' component={CharacterSheet}/>
                        <Route exact path='/character/open' component={OpenCharacter}/>
                    </div>
                </Fragment>
            </IntlProvider>
        );
    }
}