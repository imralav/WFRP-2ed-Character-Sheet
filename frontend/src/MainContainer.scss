body {
    font: 14px "Century Gothic", Futura, sans-serif;
    margin: 20px;
}

.w-90p {
    width: 90%;
}

.w-50p {
    width: 50%;
}

.w-40p {
    width: 40%;
}

.w-33p {
    width: 33.3%;
}

.w-15p {
    width: 15%;
}

.w-10p {
    width: 10%;
}

.w-5p {
    width: 5%;
}

.padding-20px {
    padding: 20px;
}

.header-font-size {
    font-size: 2vmax;
}

.label-font-size {
    font-size: 1.75vmax;
}

.content-font-size {
    font-size: 2vmax;
}

@media (min-width: 576px) {
    .header-font-size {
        font-size: 2vmax;
    }

    .label-font-size {
        font-size: 1.75vmax;
    }

    .content-font-size {
        font-size: 2vmax;
    }
}

@media (min-width: 768px) {
    .header-font-size {
        font-size: 1.75vmax;
    }

    .label-font-size {
        font-size: 1.25vmax;
    }

    .content-font-size {
        font-size: 1.5vmax;
    }
}

@media (min-width: 992px) {
    .header-font-size {
        font-size: 1.5vmax;
    }

    .label-font-size {
        font-size: 1vmax;
    }

    .content-font-size {
        font-size: 1.25vmax;
    }
}

@media (min-width: 1200px) {
    .header-font-size {
        font-size: 1.25vmax;
    }

    .label-font-size {
        font-size: 1vmax;
    }

    .content-font-size {
        font-size: 1.25vmax;
    }

}