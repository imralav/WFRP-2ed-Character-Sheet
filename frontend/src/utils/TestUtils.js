import React, {Fragment} from 'react';

import en from 'react-intl/locale-data/en';
import {addLocaleData, IntlProvider} from 'react-intl';
import renderer from 'react-test-renderer';

import {BrowserRouter as Router, Route} from 'react-router-dom';

addLocaleData(en);

class ComponentBuilder {
    constructor(component) {
        this.component = component;
    }

    withIntl() {
        this.component = (
            <IntlProvider locale="en" messages={{}}>
                <Fragment>
                {this.component}
                </Fragment>
            </IntlProvider>
        );
        return this;
    }

    withRouter() {
        this.component = (
            <Router>
                <Fragment>
                    {this.component}
                </Fragment>
            </Router>
        );
        return this;
    }

    build() {
        return renderer.create(this.component);
    }
}

export default class TestUtils {
    static buildComponent(component) {
        return new ComponentBuilder(component);
    }
}