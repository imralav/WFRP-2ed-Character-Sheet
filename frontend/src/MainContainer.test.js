import React from 'react';
import MainContainer from './MainContainer';
import TestUtils from './utils/TestUtils';
import Header from './header';

describe('MainContainer', () => {
    let component;
    let componentRoot;
    let tree;
    beforeEach(() => {
        component = TestUtils.createComponentWithDefaultIntlAndRouter(
            <MainContainer/>
        );
        componentRoot = component.root;
        tree = component.toJSON();
    });
    it('should match previous snapshot', () => {
        expect(tree).toMatchSnapshot();
    });
    it('should start with polish locale', () => {
        expect(componentRoot.findByType(MainContainer).instance.state.locale).toEqual('pl');
    });
    describe('after locale is changed', () => {
        const NEW_LOCALE = 'en';
        beforeEach(() => {
            componentRoot.findByType(Header).props.changeLocale(NEW_LOCALE);
        });
        it('should match snapshot', () => {
            expect(tree).toMatchSnapshot();
        });
        it('should change locale', () => {
            expect(componentRoot.findByType(MainContainer).instance.state.locale).toEqual(NEW_LOCALE);
        });
    });
});
