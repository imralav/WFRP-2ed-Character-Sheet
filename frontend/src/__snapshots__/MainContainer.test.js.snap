// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer after locale is changed should match snapshot 1`] = `
Array [
  <div
    className="container-fluid"
>
    <div
        className="row header"
    >
        <div
            className="col-3"
        >
            <a
                href="/"
                onClick={[Function]}
            >
                <span>
                    Start
                </span>
            </a>
        </div>
        <div
            className="col-9 text-right"
        >
            <a
                className="localeChangeButton"
                onClick={[Function]}
            >
                EN
            </a>
            <a
                className="localeChangeButton"
                onClick={[Function]}
            >
                PL
            </a>
        </div>
    </div>
    <div
        className="row"
    >
        <div
            className="col-12 col-sm col-md-5"
        >
            <a
                className="modeSelectorTile text-center border border-dark rounded d-inline-flex flex-column"
                href="/character"
                onClick={[Function]}
            >
                <div>
                    <img
                        alt=""
                        src="create-character.png"
                    />
                </div>
                <div>
                    <span>
                        <span>
                            Stwórz postać
                        </span>
                    </span>
                </div>
            </a>
        </div>
        <div
            className="col-12 col-sm offset-md-2 col-md-5 text-right"
        >
            <a
                className="modeSelectorTile text-center border border-dark rounded d-inline-flex flex-column"
                href="/character/open"
                onClick={[Function]}
            >
                <div>
                    <img
                        alt=""
                        src="edit-character.png"
                    />
                </div>
                <div>
                    <span>
                        <span>
                            Wykorzystaj istniejącą postać
                        </span>
                    </span>
                </div>
            </a>
        </div>
    </div>
</div>,
  ";",
]
`;

exports[`MainContainer should match previous snapshot 1`] = `
Array [
  <div
    className="container-fluid"
>
    <div
        className="row header"
    >
        <div
            className="col-3"
        >
            <a
                href="/"
                onClick={[Function]}
            >
                <span>
                    Start
                </span>
            </a>
        </div>
        <div
            className="col-9 text-right"
        >
            <a
                className="localeChangeButton"
                onClick={[Function]}
            >
                EN
            </a>
            <a
                className="localeChangeButton"
                onClick={[Function]}
            >
                PL
            </a>
        </div>
    </div>
    <div
        className="row"
    >
        <div
            className="col-12 col-sm col-md-5"
        >
            <a
                className="modeSelectorTile text-center border border-dark rounded d-inline-flex flex-column"
                href="/character"
                onClick={[Function]}
            >
                <div>
                    <img
                        alt=""
                        src="create-character.png"
                    />
                </div>
                <div>
                    <span>
                        <span>
                            Stwórz postać
                        </span>
                    </span>
                </div>
            </a>
        </div>
        <div
            className="col-12 col-sm offset-md-2 col-md-5 text-right"
        >
            <a
                className="modeSelectorTile text-center border border-dark rounded d-inline-flex flex-column"
                href="/character/open"
                onClick={[Function]}
            >
                <div>
                    <img
                        alt=""
                        src="edit-character.png"
                    />
                </div>
                <div>
                    <span>
                        <span>
                            Wykorzystaj istniejącą postać
                        </span>
                    </span>
                </div>
            </a>
        </div>
    </div>
</div>,
  ";",
]
`;
